src/gui/DatabaseTabWidget.cpp

```diff
@@ -713,5 +714,9 @@ void DatabaseTabWidget::performGlobalAutoType()
         }
     }
 
-    autoType()->performGlobalAutoType(unlockedDatabases);
+    if (unlockedDatabases.count() == 0) {
+        indexDatabaseManagerStruct(0).dbWidget->showUnlockDialog();
```

Comment:
Will be very unhappy if there is no database tab.
You'll want to iterate over the databases and show the dialog for the first database that is actually locked.